bin/node/cli/tests/websocket_server.rs

```diff
@@ -0,0 +1,427 @@
+#![cfg(not(target_os = "unknown"))]
+#![cfg_attr(docsrs, doc(cfg(all(feature = "std", not(target_os = "unknown")))))]
+
+use async_std::net::{TcpListener, TcpStream};
+use core::{fmt, pin::Pin, str};
+use futures::{channel::mpsc, prelude::*};
+use soketto::handshake::{server::Response, Server};
+use std::{io, net::SocketAddr};
+
+/// Configuration for a [`WsServer`].
+pub struct Config {
+	/// IP address to try to bind to.
+	pub bind_address: SocketAddr,
+
+	/// Maximum size, in bytes, of a frame sent by the remote.
+	///
+	/// Since the messages are entirely buffered before being returned, a maximum value is
+	/// necessary in order to prevent malicious clients from sending huge frames that would
+	/// occupy a lot of memory.
+	pub max_frame_size: usize,
+
+	/// Number of pending messages to buffer up for sending before the socket is considered
+	/// unresponsive.
+	pub send_buffer_len: usize,
+
+	/// Pre-allocated capacity for the list of connections.
+	pub capacity: usize,
+}
+
+/// Identifier for a connection with regard to a [`WsServer`].
+///
+/// After a connection has been closed, its [`ConnectionId`] might be reused.
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct ConnectionId(usize);
+
+/// WebSockets listening socket and list of open connections.
+pub struct WsServer<T> {
```

Comment:
I don't understand why we need to distinguish between different connections in this test.

Reply:
I just copy-pasted the whole thing. I don't expect this module to stay. When this module will be released I will use the released one.